examples/infra/mod.rs

```diff
@@ -342,296 +581,152 @@ impl<
-            node_index,
-            config: HotShotConfig { total_nodes, .. },
-            ..
-        } = self.get_config();
-
-        let size = mem::size_of::<TYPES::Transaction>();
-        let adjusted_padding = if padding < size { 0 } else { padding - size };
-        let mut txns: VecDeque<TYPES::Transaction> = VecDeque::new();
-        let state = hotshot.get_state().await;
-
-        // This assumes that no node will be a leader more than 5x the expected number of times they should be the leader
-        // FIXME  is this a reasonable assumption when we start doing DA?
-        let tx_to_gen = transactions_per_round * (cmp::max(rounds / total_nodes, 1) + 5);
-        error!("Generated {} transactions", tx_to_gen);
-        {
-            let mut txn_rng = rand::thread_rng();
-            for _ in 0..tx_to_gen {
-                // TODO make this u64...
-                let txn =
-                    <<TYPES as NodeType>::StateType as TestableState>::create_random_transaction(
-                        Some(&state),
-                        &mut txn_rng,
-                        padding as u64,
-                    );
-                txns.push_back(txn);
+    /// Tells the orchestrator this validator is ready to start
+    /// Blocks until the orchestrator indicates all nodes are ready to start
+    async fn wait_for_all_nodes_ready(&self, node_index: u64) -> bool {
+        let send_ready_f = |client: Client<ClientError>| {
+            async move {
+                let result: Result<_, ClientError> = client
+                    .post("api/ready")
+                    .body_json(&node_index)
+                    .unwrap()
+                    .send()
+                    .await;
+                result
             }
-        }
-
-        error!("Adjusted padding size is = {:?}", adjusted_padding);
-        let mut timed_out_views: u64 = 0;
-        let mut round = 1;
-        let mut total_transactions = 0;
-
-        let start = Instant::now();
-
-        error!("Starting hotshot!");
-        hotshot.start().await;
-        while round <= rounds {
-            debug!(?round);
-            error!("Round {}:", round);
-
-            let num_submitted = if node_index == ((round % total_nodes) as u64) {
-                tracing::info!("Generating txn for round {}", round);
-
-                for _ in 0..transactions_per_round {
-                    let txn = txns.pop_front().unwrap();
-                    tracing::info!("Submitting txn on round {}", round);
-                    hotshot.submit_transaction(txn).await.unwrap();
-                }
-                transactions_per_round
-            } else {
-                0
-            };
-            error!("Submitting {} transactions", num_submitted);
-
-            // Start consensus
-            error!("  - Waiting for consensus to occur");
-            debug!("Waiting for consensus to occur");
+            .boxed()
+        };
+        let _result: Result<(), ClientError> =
```

Comment:
Is it okay for this to return an error?

Reply:
Good catch!  This actually cannot return an error; I've updated the type to reflect this!